packages/react/src/component.tsx

```diff
@@ -17,11 +17,13 @@ type EndpointHelper<TRouter extends void | FileRouter> = void extends TRouter
 export function UploadButton<TRouter extends void | FileRouter = void>(props: {
   endpoint: EndpointHelper<TRouter>;
   onClientUploadComplete?: () => void;
+  onUploadError?: (error: Error) => void;
 }) {
   const { startUpload, isUploading, permittedFileInfo } =
     useUploadThing<string>({
```

Comment:
Can we just pass `props` here?

Reply:
No, because EndpointHelper type not resolving to string. Maybe this type can be refined to make this work?